File: Kms.Security.Portal/app/nodak/common/enterprisePositionPost/enterprisePositionPost.ts
module nodak.common.models {

    export class EnterprisePositionPost extends nodak.models.ModelBase {

        constructor() {
            super(enums.SubSystems.Common, "EnterprisePositionPost");
        }
        AddBussinessModelValidation() {

        }
        private id: any
        set Id(value)
        { this.id = value; }
        get Id()
        { return this.id; }

        private title: string;
        get Title() {
            return this.title;
        }
        set Title(value) {
            this.title = value;
        }

        private description: string;
        get Description() {
            return this.description;
        }
        set Description(value) {
            this.description = value;
        }

        private enterprisePositionId: string;
        get EnterprisePositionId() {
            return this.enterprisePositionId;
        }
        set EnterprisePositionId(value) {
            this.enterprisePositionId = value;
        }

        private enterprisePosition: EnterprisePosition;
        set EnterprisePosition(value) {
            this.enterprisePosition = value;
        }
        get EnterprisePosition() {
            return this.enterprisePosition;
        }

        private permissions: Array<models.Permission>;
        get Permissions() {
            return this.permissions;
        }
        set Permissions(value) {
            this.permissions = value;
        }
    }
}

File: Kms.Security.Common/Domain/EnterprisePositionPost.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Kms.Security.Common.Domain
{
    public class EnterprisePositionPost : IDomainBase<Guid>
    {
        public EnterprisePositionPost()
        {

        }
        public EnterprisePositionPost(Guid id, string title,Guid enterprisePositionId, string description)
        {
            Id = id;
            Title = title;
            EnterprisePositionId = enterprisePositionId;
            Description = description;
        }
        public Guid Id { protected set; get; }

        public bool IdIsEmpty()
        {
            return Id == Guid.Empty;
        }

        public void SetNewId()
        {
            Id = Guid.NewGuid();
        }
        public string Title { get;  set; }
        public EnterprisePosition EnterprisePosition { get;  set; }
        public Guid EnterprisePositionId { get;  set; }
        public string Description { get;  set; }

        public List<Permission> Permissions { get;  set; }
        public void SetPermissionList(List<Permission> permissionList)
        {
            this.Permissions = permissionList;
        }
    }
}
